### DISCUSSION TITLE
Istio default load balancing issues

### DISCUSSION THREAD

**Discussion by SaiKiranAre** [istio/istio#46586] [General]
Hi There,

I am using istio version 1.17.5

 I have two deployments for grpc service(TLS) and have common meta data label. And defined ClusterIP service route to those deployments using common label.

We enabled service mesh and MTLS enabled between the services. Things works fine, at some point of time. Suddenly the traffic is going to only one pod.

I ran ```istioctl pc endpoint``` command to see the end point health. It is listing both the endpoints and healthy.

How can I debug and fix these kind of problems?

**Comment by SaiKiranAre** [comment]
Envoy proxy provides least_request LB with choice count. Is there any way to set this in istio?

Ref: https://github.com/envoyproxy/envoy/pull/4593

**Reply by bnogas** [reply]
I think it's not exposed in istio. Such feature would be nice to have:

In istio we just have an enum:
https://github.com/istio/istio/blob/0dd4d9e4e4b1e4fa83adca9c9167de69bfaeda9c/pilot/pkg/networking/core/cluster_traffic_policy.go#L265C12-L265C21

So there is no way to provide this config:
https://github.com/envoyproxy/envoy/blob/78a88415a4e1ba9114a06d1d0a1491398e17e995/source/extensions/load_balancing_policies/least_request/least_request_lb.h#L44
